app/models/loan.rb

```diff
@@ -1,2 +1,125 @@
+# == Schema Information
+#
+# Table name: loans
+#
+#  id                  :bigint           not null, primary key
+#  current_balance     :float
+#  original_amount     :float
+#  interest_rate       :float
+#  periods_in_year     :integer
+#  user_id             :integer
+#  loan_name           :string
+#  monthly_min_payment :float
+#  schedule_hash       :jsonb
+#  adj_schedule_hash   :jsonb
+#  created_at          :datetime         not null
+#  updated_at          :datetime         not null
+#
 class Loan < ApplicationRecord
+  has_many :adjustments, :dependent => :destroy
+  belongs_to :user
+    
+    validates :interest_rate, :numericality => { :greater_than => 0 }
+    validates :interest_rate, :presence => true
+    validates :current_balance, :numericality => { :greater_than_or_equal_to => 0 }
+    validates :current_balance, :presence => true
+    validates :original_amount, :numericality => { :greater_than => 0 }
+    validates :monthly_min_payment, :presence => true
+    validates :monthly_min_payment, :numericality => { :greater_than => 0 }
+    validates :periods_in_year, :presence => true
+    validates :periods_in_year, :numericality => { :greater_than => 0 }
+    
+    def pay_schedule
+        i=0;
+        end_balance = self.current_balance;
+        pmt_amt = self.monthly_min_payment;
+        interest_rate =  self.interest_rate.to_f / 100;
+        periods = self.periods_in_year;
```

Comment:
I need to dive into this a little more, but I actually do need to explicitly call out `self` or my calculations in my loops don't work correctly.  Specifically, I am finding that not calling out `self.interest_rate.to_f / 100` is causing the issue.  This method get's called within the following loop currently in my Loan Index Controller.  

```ruby
      @loans.each_with_index do |loan , index|
        pmt_schedule = loan.pay_schedule;  
        pmt_schedule_adj = loan.adj_pay_schedule(loan.adjustments);
        @pmt_schedule_chart[index] =  pmt_schedule.map{|t| [ t["pmt_no"], t["beg_balance"] ] };
        @adj_pmt_schedule_chart[index] = pmt_schedule_adj.map{|t| [ t["pmt_no"], t["beg_balance_adj"] ] };
        @total_interest += pmt_schedule[-1].fetch("cum_int");
        @total_adj_interest += pmt_schedule_adj[-1].fetch("cum_int_adj");
        @pay_off_period[index] = pmt_schedule.length;
        @pay_off_period_adj[index] = pmt_schedule_adj.length;
      end
```

I'll try and determine exactly why, but if something jumps out to you as well let me know, or else I'll respond here when I figure out why.

Reply:
@jelaniwoods I think I need to keep these `self` 's in here since this method is also called in another loop.  I get errors when I run this without them... still digging into the root cause.